doc/releases/release-notes-4.1.rst

```diff
@@ -82,6 +82,8 @@ Removed APIs and options
 * The :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE` option has been removed
   after being deprecated in favor of :kconfig:option:`CONFIG_PM_DEVICE_SYSTEM_MANAGED`.
 
+* The ``z_pm_save_idle_exit()`` PM API function has been remove.
```

Comment:
```suggestion
* The ``z_pm_save_idle_exit()`` PM API function has been removed.
```

Reply:
it should not have been deprecated in the first place actually, ust dropped...

Reply:
The deprecated function was part of the public header and was a very old function. I was concerned that one could have been used out of tree, which is why I deprecated it.

Reply:
Btw, I can drop the release notes commit if you want.